### PR TITLE
Update release workflow to v3

### LINKED ISSUE #4294: Update release workflow to v3
https://jira.sonarsource.com/browse/BUILD-909?focusedCommentId=246823&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-246823

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #4294

### CODE REVIEW

**Comment by andrei-epure-sonarsource** [comment]
FYI I am updating the parent pom in #4328

File: pom.xml [hunk 1]
```diff
@@ -78,5 +78,30 @@
         <maven.compiler.release>8</maven.compiler.release>
       </properties>
     </profile>
+    <profile>
```
Comment (andrei-epure-sonarsource) [inline pom.xml:81]: @costin-zaharia-sonarsource is this really needed if the parent pom is updated?
I looked at e,g, sonar-php whcih got released, it doesn't have this

Comment (andrei-epure-sonarsource) [reply pom.xml:81]: this is already in the parent pom https://github.com/SonarSource/parent-oss/blob/e7e21714eee9a861d2371b174f7ab804f7a79f06/pom.xml#L577

Please rebase on master and remove this

Comment (costin-zaharia-sonarsource) [reply pom.xml:81]: The agent runs as a `NetworkService` and for this user we don't have a default `.m2` folder from where the script can take the signing key.

In order to fix this we need to override the profile and set the path to the key file.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -280,10 +286,11 @@ stages:
         env:
           ARTIFACTORY_DEPLOY_USERNAME: $(ARTIFACTORY_DEPLOY_USERNAME)
           ARTIFACTORY_DEPLOY_PASSWORD: $(ARTIFACTORY_DEPLOY_PASSWORD)
+          PGP_PASSPHRASE: $(PGP_PASSPHRASE)
```
Comment (costin-zaharia-sonarsource) [inline azure-pipelines.yml:289]: This is not propagated by default. We have to copy it by hand (similar with `ARTIFACTORY_DEPLOY_USERNAME` and `ARTIFACTORY_DEPLOY_PASSWORD`).

**Review by tomverin — APPROVED** [review]

**Pull request merged**